beacon-chain/forkchoice/protoarray/store.go

```diff
@@ -405,6 +412,23 @@ func (s *Store) applyWeightChanges(ctx context.Context, justifiedEpoch, finalize
 
 		nodeDelta := delta[i]
 
+		// If we have a node where the proposer boost was previously applied,
```

Comment:
This is incorrect. We need to account for our ancestors. Example:
```
A -> B -> C ------> E
             \->D
 ```
E arrives timely, this section of code applies proposer boost score to it. Its ancestors A, B, and C should all have the same proposer boost score applied to them

Reply:
Confirmed we backpropagate to ancestors below